test/contact_test.dart

```diff
@@ -22,16 +22,31 @@ void main() {
         return c;
       }
     });
-    final newContact = Contact()
-      ..name = Name(first: 'John', last: 'Doe')
-      ..phones = [
-        Phone('555-123-4567'),
-        Phone('555-999-9999', label: PhoneLabel.work)
-      ];
+    final newContact = Contact(name: Name(first: 'John', last: 'Doe'), phones: [
+      Phone('555-123-4567'),
+      Phone('555-999-9999', label: PhoneLabel.work)
+    ]);
+
     final returnedContact = await newContact.insert();
     expect(returnedContact.displayName, 'John Doe');
     expect(returnedContact.phones.length, 2);
   });
+
+  test('When using copyWith() with attributes, new object is not a reference',
+      () {
+    final Contact contact1 = Contact(id: "1", displayName: "x");
+    final Contact contact2 = contact1.copyWith(displayName: "y");
+
+    expect(contact1.displayName == contact2.displayName, false);
+  });
+  test(
+      'When using copyWith() with attributes, validate equality '
+      'operator function detects objects difference', () {
+    final Contact contact1 = Contact(id: "1", displayName: "x");
+    final Contact contact2 = contact1.copyWith(id: "2");
+
+    expect(contact1 == contact2, false);
+  });
```

Comment:
You should add tests for copy with `null`. Something like
```dart
expect(contactWithPhoto.copyWith(photo: null).photo, isNull);
```
Currently this would fail.